client/my-sites/email/titan-set-up-thank-you/index.tsx

```diff
@@ -61,69 +65,102 @@ const TitanSetUpThankYou = ( {
 		}
 	);
 
-	const titanThankYouSection = {
-		sectionKey: 'titan_whats_next',
-		sectionTitle: translate( 'What’s next?' ),
-		nextSteps: [
+	let nextSteps = [
+		{
+			stepKey: 'titan_whats_next_view_inbox',
+			stepTitle: translate( 'Access your inbox' ),
+			stepDescription: translate( 'Access your email from anywhere with our webmail.' ),
+			stepCta: (
+				<FullWidthButton
+					href={ getTitanEmailUrl(
+						titanAppsUrlPrefix,
+						emailAddress,
+						false,
+						`${ window.location.protocol }//${ window.location.host }${ inboxPath }`
+					) }
+					primary
+					onClick={ () => {
+						recordEmailAppLaunchEvent( {
+							provider: 'titan',
+							app: 'webmail',
+							context: 'checkout-thank-you',
+						} );
+					} }
+				>
+					{ translate( 'Go to Inbox' ) }
+				</FullWidthButton>
+			),
+		},
+		{
+			stepKey: 'titan_whats_next_get_mobile_app',
+			stepTitle: translate( 'Get mobile app' ),
+			stepDescription: translate(
+				"Access your email on the go with Titan's Android and iOS apps."
+			),
+			stepCta: (
+				<FullWidthButton
+					href={ titanControlPanelUrl }
+					target="_blank"
+					onClick={ () => {
+						recordEmailAppLaunchEvent( {
+							provider: 'titan',
+							app: 'app',
+							context: 'checkout-thank-you',
+						} );
+					} }
+				>
+					{ translate( 'Get app' ) }
+					<Gridicon className="titan-set-up-thank-you__icon-external" icon="external" />
+				</FullWidthButton>
+			),
+		},
+		{
+			stepKey: 'titan_whats_next_manage_email',
+			stepTitle: translate( 'Manage your email' ),
+			stepDescription: translate(
+				'Add or delete mailboxes, migrate existing emails, configure a catch-all email, and much more.'
+			),
+			stepCta: (
+				<FullWidthButton href={ emailManagementPath }>{ translate( 'Manage' ) }</FullWidthButton>
+			),
+		},
+	];
+
+	if ( emailNeedsSetup ) {
+		nextSteps = [
 			{
-				stepKey: 'titan_whats_next_view_inbox',
-				stepTitle: translate( 'Access your inbox' ),
-				stepDescription: translate( 'Access your email from anywhere with our webmail.' ),
+				stepKey: 'titan_whats_next_setup_mailbox',
+				stepTitle: translate( 'Set up your email' ),
+				stepDescription: translate( 'Complete the setup by choosing your email address.' ),
 				stepCta: (
 					<FullWidthButton
-						href={ getTitanEmailUrl(
-							titanAppsUrlPrefix,
-							emailAddress,
-							false,
-							`${ window.location.protocol }//${ window.location.host }${ inboxPath }`
-						) }
+						href={ emailManagementTitanSetUpMailbox( selectedSiteSlug ?? '', domainName ) }
```

Comment:
The first time I clicked the CTA, `/email/:domain/titan/set-up-mailbox/:site` redirected to `/email/:domain/purchase/:site`. It appears that the newly purchased domain wasn't updated in the `state`, thus the `hastTitanMailWithUs()` returns false and we end up getting redirected:
https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/email/titan-set-up-mailbox/index.tsx#L52

If I navigate back to the "Thank you" page, and click the CTA again, it works as expected.